testsuite/test_wbemconnection_mock.py

```diff
@@ -4367,3 +4372,166 @@ def test_invokemethod(self, conn, ns, desc, inputs, exp_output,
                 print('exc.status_code_name %s cim_err_code %s' %
                       (exc.status_code_name, cim_err_code_name))
                 assert exc.status_code_name == cim_err_code_name
+
+
+@pytest.fixture()
+def test_schema_dir():
+    """
+    Fixture defines the test_schema_dir in the testsuite and also
+    removes the directory if it exists at the end of each test.
+    """
+    schema_dir = os.path.join(SCRIPT_DIR, 'test_schema')
+    yield schema_dir
+
+    # Executes at the close of each test function
+    if os.path.isdir(schema_dir):
+        shutil.rmtree(schema_dir)
+
+
+class TestDMTFSchema(object):
+    """
+    Test the DMTFSchema class in pywbem_mock.  Since we do not want to always
+    be downloading the DMTF schema for CI testing, we will generally skip the
+    first test.
+    """
+
+    def test_schema_final_load(self, test_schema_dir):
+        # pylint: disable=no-self-use
+        """
+        Test the DMTFSchema class and its methods to get a schema from the
+        DMTF, expand it, and to create a partial mof schema definition for
+        compilation.
+        """
+        schema = DMTFSchema((2, 49, 0), test_schema_dir, verbose=True)
```

Comment:
I think this test should ensure that before DMTFSchema is expected to create all the file and directory artefacts, that they do not exist.

Reply:
TODO